subworkflows/local/prepare_input.nf

```diff
@@ -0,0 +1,64 @@
+nextflow.enable.dsl = 2
+
+import org.yaml.snakeyaml.Yaml
+
+workflow PREPARE_INPUT {
+
+    take:
+    ch_input
+
+    main:
+
+    ch_input.map { file -> readYAML( file ) }
+        .multiMap { file ->
+            samples_ch: file.samples
+        }
+        .set { ymlfile }
+
+    ymlfile.samples_ch
```

Comment:
```suggestion
    ch_input.map { file -> readYAML( file ).samples }
```
Better suggestion to before. 

Acknowledging where this was adapted from would be appreciated.

Reply:
Sure, when the pipeline will be published all credits will be acknowledged on the main github page under 'Credits' section

Reply:
I'm more talking about just referencing that this was adapted from https://github.com/mahesh-panchal/genomeassembler/blob/230fa883621e45ce3d1002eb615bd5a01a68e59b/subworkflows/local/prepare_input.nf rather than crediting nf-core in the credits.